src/engine/blocks.js

```diff
@@ -0,0 +1,206 @@
+/**
+* @fileoverview
+ * Store and mutate the VM block representation,
+ * and handle updates from Scratch Blocks events.
+ */
+
+ function Blocks () {
+     /**
+      * All blocks in the workspace.
+      * Keys are block IDs, values are metadata about the block.
+      * @type {Object.<string, Object>}
+      */
+     this._blocks = {};
+
+     /**
+      * All stacks in the workspace.
+      * A list of block IDs that represent stacks (first block in stack).
+      * @type {Array.<String>}
+      */
+     this._stacks = [];
+ }
+
+/**
+ * Provide an object with metadata for the requested block ID.
+ * @param {!string} blockId ID of block we have stored.
+ * @return {?Object} Metadata about the block, if it exists.
+ */
+ Blocks.prototype.getBlock = function (blockId) {
+     return this._blocks[blockId];
+ };
+
+/**
+ * Get all known top-level blocks that start stacks.
+ * @return {Array.<string>} List of block IDs.
+ */
+ Blocks.prototype.getStacks = function () {
+     return this._stacks;
+ };
+
+ /**
+  * Get the next block for a particular block
+  * @param {?string} id ID of block to get the next block for
+  * @return {?string} ID of next block in the sequence
+  */
+ Blocks.prototype.getNextBlock = function (id) {
+     if (typeof this._blocks[id] === 'undefined') return null;
+     return this._blocks[id].next;
```

Comment:
That sounds good in the long run. Can't quite do it yet because of https://github.com/google/blockly/issues/336